src/Microsoft.AspNetCore.Server.IISIntegration/WebHostBuilderIISExtensions.cs

```diff
@@ -39,6 +41,21 @@ public static IWebHostBuilder UseIISIntegration(this IWebHostBuilder hostBuilder
                 return hostBuilder;
             }
 
+            // Check if in process
+            if (NativeMethods.is_ancm_loaded())
+            {
+                // TODO put this in options and use path.
+                var applicationPath = NativeMethods.http_get_application_full_path();
+                hostBuilder.UseContentRoot(applicationPath);
+                return hostBuilder.ConfigureServices(services =>
+                {
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                    {
+                        services.AddSingleton<IServer, IISHttpServer>();
```

Comment:
We need a way to avoid ordering issues with UseKestrel. IStartupFilter isn't going to work, that doesn't support services. We may need to change Kestrel to TryAdd.

Reply:
Yea this is an interesting issue.